app/controllers/users_controller.rb

```diff
@@ -143,13 +143,24 @@ def remove_identity
   def onboarding_update
     if params[:user]
       sanitize_user_params
-      permitted_params = %i[summary location employment_title employer_name last_onboarding_page]
-      current_user.assign_attributes(params[:user].permit(permitted_params))
+      permitted_user_params = %i[last_onboarding_page]
+      current_user.assign_attributes(params[:user].permit(permitted_user_params))
       current_user.profile_updated_at = Time.current
+      if current_user.save
+        success = true
```

Comment:
I think we don't need this local variable. You can rewrite the next conditional instead and directly use the value of `update_result.success?`:

```ruby
    if current_user.save && params[:profile]
      updated_result = Profiles::Update.call(current_user, { profile: profile_params })
    end
    
    current_user.saw_onboarding = true
    authorize User
    render_update_response(update_result.success?)
```

Note 1: This behaves differently from your original code. In your code a failed call to `Profiles::Update` would not have resulted in `success` being `false`. I think the way it should have looked originally is

```ruby
    if params[:profile] && success == true
      update_result = Profiles::Update.call(current_user, { profile: params[:profile].permit(Profile.attributes!) })

      success = update_result.success?
```

The way the code currently is it would have set `success` to true again (which it already was if we are inside this branch due to the outer conditional), but never to `false`. Please let me know if I misread something.

Note 2: I also added a `profile_params` method call here and would encourage extracting it for readability instead of keeping `params[:profile].permit(Profile.attributes!)` inline.

Reply:
@citizen428 thank you for taking the time to do this write up. Its been really useful and I've refactored accordingly.

Reply:
```
    if current_user.save && params[:profile]
      update_result = Profiles::Update.call(current_user, { profile: profile_params })
    end
    
    current_user.saw_onboarding = true
    authorize User
    render_update_response(update_result.success?)
```
Just hit a snag and realized that if `current_user.save && params[:profile]` evaluates to false then update_result will not be defined 3 lines down.I didnt remember at the time, but that was the reason I may have defined the success var. I changed this to be  `render_update_response(update_result&.success?)`.

Reply:
Ah, yes, correct @Ridhwana. I didn't try out this code, I just wrote it directly in here, but `&.` seems like an ok solution.